src/utils/more_lodash.js

```diff
@@ -0,0 +1,21 @@
+import _ from 'lodash';
```

Comment:
@armenzg @camd   Where can I access the exports of lodash (or any other module)?  I am spending too much time trying to guess what a module exports.

For example, import `isEqual from 'lodash/isEqual'` works, while `import chain from 'lodash/chain'` does not. But `import { chain } from 'lodash/chain'` seems to work, but neither `import { mixin } from 'lodash/mixin'` nor `import mixin from 'lodash/mixin'` work.

Reply:
Also, it seems lodash is 24kb (https://lodash.com/), given the low number of customers, and given their known high quality connection, plus the fact the chart data takes longest to arrive, I am dubious that this library is worth optimizing.

Reply:
> I think in some cases you can get rid of lodash with a bit of map & reduce.

Yes, *all* functional expressions can be rephrased in terms of `map` and `reduce`. The Boolean operators AND and OR can also be replaced with NAND; should we get rid of AND and OR too?  

Operations that are commonly used should be named and implemented.  Some examples are: filtering, groupby, joining-tables, transposing-a-grid, selecting-and-renaming-a-subset-of-columns.  These are common operations in the data management space, and exist as primitive operations there. Lodash does a reasonable job of bringing those common operations to Javascript. If we do not use Lodash, then my next option is to re-implement Lodash functionality in `utils`.  (I would be happy to do that, since Lodash is clunky in many ways that either you or I have already noticed).

If we use common functional programming methods (aka common query operators), and we flatten our data structures, the code will get simpler.  `reduce()` should not be used; it promotes complicated data structures because it can do "anything": `reduce()` in functional programming is like `goto` in structured programming: The flexibility of the operator promotes complicated code.

Reply:
I beg to disagree that `reduce` makes anything complicated. We both are entitled to our opinions.

> @armenzg @camd Where can I access the exports of lodash (or any other module)? I am spending too much time trying to guess what a module exports.
> 
> For example, import `isEqual from 'lodash/isEqual'` works, while `import chain from 'lodash/chain'` does not. But `import { chain } from 'lodash/chain'` seems to work, but neither `import { mixin } from 'lodash/mixin'` nor `import mixin from 'lodash/mixin'` work.

I would install the packages and look under `./node_modules/lodash`.
Sometimes the README files of the packages is sufficient to determine this.



> Also, it seems lodash is 24kb (https://lodash.com/), given the low number of customers, and given their known high quality connection, plus the fact the chart data takes longest to arrive, I am dubious that this library is worth optimizing.

It's not a matter of how big lodash is. It's a matter if we're going to keep following a standard practice (not just within @mozilla-frontend-infra but across the web). The standard practice is that we import explicitely since the world we live in is not 100% tree shakeable. Perhaps in the future we won't have to worry but right now we do.

We simply don't have to spend time evaluating the cost of being explicit or not. We simply require to always be explicit. If you want to discuss the practices we follow feel free to start a discussion outside of the PR. It will make it easier to discuss.

Reply:
@armen  I removed `_.chain` and rewrote it as `frum` so the code reads nicer.  I am open for different naming.